services/github/github-tag.service.js

```diff
@@ -99,7 +109,13 @@ class GithubTag extends GithubAuthV4Service {
     })
   }
 
-  static getLatestTag({ tags, sort, includePrereleases }) {
+  static getLatestTag({ tags, sort, includePrereleases, prefix }) {
+    if (prefix) {
+      const tagsWithPrefix = tags.filter(tag => tag.startsWith(prefix))
+      if (tagsWithPrefix.length > 0) {
+        tags = tagsWithPrefix
+      }
+    }
```

Comment:
My guess is that you did this given what you said about the 100 tag limit given the context of the current query. However, I believe if we can get the query right and allow GitHub to only return results that match that prefix, then we can handle things more directly here. I.e. if we ask GitHub to give us only the tags that are prefixed with `foo-` and GitHub tells us there are none, then we can and should throw an error here (a `NotFound`) which we could do safely

Reply:
Agreed :+1:  Lets throw a `NotFound()` if there are 0 matching tags

Reply:
Done, thanks!